src/EnergyPlus/HeatBalanceSurfaceManager.cc

```diff
@@ -307,42 +307,71 @@ void InitSurfaceHeatBalance(EnergyPlusData &state)
-                    Surface(SurfNum).ViewFactorGroundIR = state.dataSurface->SurroundingSurfsProperty(SrdSurfsNum).GroundViewFactor;
-                } else if (state.dataSurface->SurroundingSurfsProperty(SrdSurfsNum).SkyViewFactor >= 0 &&
-                           state.dataSurface->SurroundingSurfsProperty(SrdSurfsNum).GroundViewFactor < 0) {
+                    Surface(SurfNum).ViewFactorSkyIR = SurfsSkyViewFactor;
+                    Surface(SurfNum).ViewFactorGroundIR = GroundSurfsViewFactor;
+                } else if (SurfsSkyViewFactor >= 0 && !IsGroundViewFactorSet) {
                     // If only sky view factor defined, ground view factor = 1 - all other defined view factors.
-                    Surface(SurfNum).ViewFactorSkyIR = state.dataSurface->SurroundingSurfsProperty(SrdSurfsNum).SkyViewFactor;
+                    Surface(SurfNum).ViewFactorSkyIR = SurfsSkyViewFactor;
                     Surface(SurfNum).ViewFactorGroundIR = 1 - SrdSurfsViewFactor;
-                } else if (state.dataSurface->SurroundingSurfsProperty(SrdSurfsNum).SkyViewFactor < 0 &&
-                           state.dataSurface->SurroundingSurfsProperty(SrdSurfsNum).GroundViewFactor >= 0) {
+                    if (GndSurfsNum > 0) {
+                        SetGroundViewFactorObject = true;
+                        state.dataSurface->GroundSurfsProperty(GndSurfsNum).IsGroundViewFactorSet = true;
+                    }
+                } else if (SurfsSkyViewFactor < 0 && IsGroundViewFactorSet) {
                     // If only ground view factor defined, sky view factor = 1 - all other defined view factors.
-                    Surface(SurfNum).ViewFactorGroundIR = state.dataSurface->SurroundingSurfsProperty(SrdSurfsNum).GroundViewFactor;
+                    Surface(SurfNum).ViewFactorGroundIR = GroundSurfsViewFactor;
                     Surface(SurfNum).ViewFactorSkyIR = 1 - SrdSurfsViewFactor;
+                    if (SrdSurfsNum > 0) {
+                        state.dataSurface->SurroundingSurfsProperty(SrdSurfsNum).SkyViewFactor = Surface(SurfNum).ViewFactorSkyIR;
+                    }
                 } else {
                     // If neither ground or sky view factor define, continue to use the original proportion.
                     Surface(SurfNum).ViewFactorSkyIR *= 1 - SrdSurfsViewFactor;
                     Surface(SurfNum).ViewFactorGroundIR *= 1 - SrdSurfsViewFactor;
+                    if (SrdSurfsNum > 0) {
+                        state.dataSurface->SurroundingSurfsProperty(SrdSurfsNum).SkyViewFactor = Surface(SurfNum).ViewFactorSkyIR;
+                    }
+                    if (GndSurfsNum > 0) {
+                        SetGroundViewFactorObject = true;
+                        state.dataSurface->GroundSurfsProperty(GndSurfsNum).IsGroundViewFactorSet = true;
+                    }
+                }
+                if (SetGroundViewFactorObject) {
+                    ReSetGroundSurfacesViewFactor(state, SurfNum);
```

Comment:
If the ground view factor field is blank and the ground view factor recalculated based on view factor identity, then the calculated view factor needed to be passed back to the new object. The function `ReSetGroundSurfacesViewFactor()` updates the new object variables.